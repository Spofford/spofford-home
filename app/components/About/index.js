import React, { Component } from 'react';
import HubspotForm from 'react-hubspot-form'
const ReactMarkdown = require('react-markdown')
import style from "./style.css"
import * as contentful from 'contentful'
import { connect } from "react-redux"
import cssModules from 'react-css-modules'
import { fetchContent} from "../../redux/actions"

import { default as Header } from "../Header"

export class About extends React.Component {

  state = {
    model: {}
  }

  componentDidMount() {
    this.props.getContent('wKhyV2HRBeSgK4UcOoiEi')
  }

  componentDidUpdate(prevProps) {
    if (prevProps.content != this.props.content) {
      this.setState({
        model: this.props.content.items[0].fields
      })
    }
  }

  render() {
    return (
      <div className="about">
        <Header headerStart={true} />
        <div className="hero">
          <div className="copy-container">
            <h2 className="head">{this.state.model.head}</h2>
            <h3 className="subhead">{this.state.model.subhead}</h3>
          </div>
        </div>
        <div className="body-container">
          <div className="copy-container">

            <ReactMarkdown source={this.state.model.bodyText} />
            <HubspotForm
               portalId='4042167'
               formId='d1e13228-c396-4338-bbe3-67cd84f53065'
               />
          </div>
        </div>
      </div>
    )
  }
}

const mapDispatchToProps = {
  getContent: fetchContent
};

const mapStateToProps = state => ({
  content: state.content
})

export default connect(mapStateToProps, mapDispatchToProps)(cssModules(About, style))
