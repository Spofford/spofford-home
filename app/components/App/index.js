import React from "react"
import { connect } from "react-redux"
import { userAuth } from "../../redux/actions"
import { withRouter } from 'react-router-dom'

export class App extends React.Component {

  componentDidMount() {
    this.props.userAuth()
    window.addEventListener('scroll', this.handleScroll);
  }

  componentDidUpdate(prevProps) {
    if (this.props.location !== prevProps.location) {
      window.scrollTo(0, 0);
    }
  }

  render() {
    return (
      <div>
        {this.props.children}
      </div>
    )
  }
}

const mapDispatchToProps = {
  userAuth: userAuth
};

const mapStateToProps = state => ({
  content: state.content
})

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(App))
//export default withRouter(App);
