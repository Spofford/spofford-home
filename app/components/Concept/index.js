import React from "react"
import * as contentful from 'contentful'
const ReactMarkdown = require('react-markdown')
import style from "./style.css"
import classNames from 'classnames';
import FontAwesome from "react-fontawesome";

export class Concept extends React.Component {
  constructor(props) {
    super(props);
    // Don't call this.setState() here!
    this.state = {
      model: this.props.concept,
      currentSketch: 0,
      allSketches: this.props.concept.fields.sketches,
      isOverlayOn: false,
      showPrev: false,
      showNext: true,
      orientation:''
    };

    this.prevSketch = this.prevSketch.bind(this);
    this.nextSketch = this.nextSketch.bind(this);
    this.toggleOverlay = this.toggleOverlay.bind(this);
  }


  client = contentful.createClient({
    space: 'cahjy08ew1qz',
    accessToken: '37c6ec31a1a6cb3f533f51fa4c4af8fee88e2f910d9879eb79b2d073ae8cc499'
  })

  componentDidMount() {
    let self = this
    this.setOrientation()
    window.addEventListener("resize", function() {
      self.setOrientation()
    });
    if (this.state.allSketches.length==1) {
      this.setState({
        showNext: false
      })
    }
  }

  setOrientation() {
    var self = this;
    if (window.innerHeight > window.innerWidth) {
      self.setState({
        orientation: 'portrait'
      })
    } else if (window.innerWidth > window.innerHeight) {
      self.setState({
        orientation: 'landscape'
      })
    }
  }

  prevSketch() {
    var currentSketch = --this.state.currentSketch
    var self = this

    if (currentSketch==0) {
      this.setState({
        showPrev: false
      })
    }
    this.setState({
      currentSketch: currentSketch,
      showNext: true
    })
  }

  nextSketch() {
    var currentSketch = ++this.state.currentSketch
    var self = this

    if (currentSketch==(this.state.allSketches.length-1)) {
      this.setState({
        showNext: false
      })
    }
    this.setState({
      currentSketch: currentSketch,
      showPrev: true
    })

  }

  toggleOverlay() {
    this.setState(prevState => ({
      isOverlayOn: !prevState.isOverlayOn
    }));
    var elem = document.getElementsByTagName("body")[0];
    if (elem.style.overflow=="hidden") {
      elem.style.overflow="auto"
    } else {
      elem.style.overflow="hidden"
    }
    if (this.state.isOverlayOn==true) {
      this.setState({
        currentSketch:0,
        showPrev: false,
        showNext: true
      })
    }
  }

  render() {
    var sketchClasses = classNames('sketch-container', this.state.orientation, {"visible": this.state.isOverlayOn});

    var chevrons = classNames({'showNext': this.state.showNext}, {'showPrev': this.state.showPrev}, 'sketch')

    return (
      <div className='concept'>
        <div onClick={this.toggleOverlay} className="tile-overlay"></div>
        <h3>{this.state.model.fields.conceptName}</h3>
        <h3>Click image to view sketches</h3>
        <img src={this.state.model.fields.primaryImage.fields.file.url} />
        <div className={sketchClasses}>
          <div className="inner-container">
            <div className="sketches-top-bar">
              <div>Slide {this.state.currentSketch + 1} of {this.state.allSketches.length}</div>
              <FontAwesome onClick={this.toggleOverlay} name='times' size='2x' />
              {this.state.isOverlayOn}
            </div>
            <div className={chevrons}>
              <img src={this.state.allSketches[this.state.currentSketch].fields.sketch.fields.file.url} />
              <div className="caption">{this.state.allSketches[this.state.currentSketch].fields.caption}</div>
              <FontAwesome onClick={this.prevSketch} name='chevron-left' size='3x' />
              <FontAwesome onClick={this.nextSketch} name='chevron-right' size='3x' />
            </div>
          </div>
        </div>
      </div>
    )
  }
}

export default Concept
