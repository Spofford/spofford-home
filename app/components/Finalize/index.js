import React from "react"
import ReactDOM from 'react-dom'
import cssModules from "react-css-modules"
import style from "./style.css"
import { default as Header } from "../Header"
import Actions from "../../redux/actions"
import { connect } from "react-redux"
import { Link, Redirect } from 'react-router-dom'
import classNames from 'classnames';
import StripeCheckout from 'react-stripe-checkout';

export class Finalize extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      publishArray: [],
      amountOwed: 0,
      // amountPaid: this.props.user.amount_paid
    }

    this.submit = this.submit.bind(this)
  }

  componentDidMount() {
    if (this.props.user.id) {
      this.props.dispatch(Actions.mySubmissions(this.props.user.id))
    }
    this.handleInputChange = this.handleInputChange.bind(this);
  }

  componentDidUpdate(prevProps) {
     if (prevProps.user.id !== this.props.user.id) {
       this.props.dispatch(Actions.mySubmissions(this.props.user.id))
     }
   }

   handleInputChange(event) {
      const target = event.target;
      const value = target.type === 'checkbox' ? target.checked : target.value;
      const name = target.name;

      const prevArray = this.state.publishArray

      if (prevArray.includes(name)) {
        var index = prevArray.indexOf(name);
        if (index > -1) {
          prevArray.splice(index, 1);
        }
        this.setState({
          publishArray: prevArray
        })
      } else {
        prevArray.push(name)
        this.setState({
          publishArray: prevArray
        })
      }

      // Calculate based on if somethign already paid... TO do
      if (this.state.publishArray.length > 0 && this.state.publishArray.length <= 3) {
        this.setState({amountOwed:2500})
      } else if (this.state.publishArray.length > 3) {
        var amount = 25
        var loop = this.state.publishArray.length - 3

        for (var i = 0; i < loop; i++ ) {
          amount = amount + 500
        }
        this.setState({amountOwed:amount})
      }

    }

   mySubmissions(props) {
     const ms = this.props.mySubmissions;
     const unapproved = ms.filter(function(thing) {
       return thing.approved != true
     })
     const unapprovedItems = unapproved.map((item) =>
     <label key={item.id}>
         <input
           name={item.id}
           type="checkbox"
           // checked={this.state.isPublishing}
           onChange={this.handleInputChange} />
           Submission #0{item.id}
       </label>
     );
     return (
       <div>
        <h2>Unpublished Submissions</h2>
        <ul>{unapprovedItems}</ul>
      </div>
     );
   }

   firstPayment = () => {
     // If publish array length and amount already paid is 0
     if (this.state.publishArray.length > 0) {
       return <div className="line-item"><span>First 3 Designs</span><span>$25</span></div>
     }
   }

   additionalPayments = () => {
     if (this.state.publishArray.length > 3) {
       var iterNumber = this.state.publishArray.length - 3
       for (var i=0;i<iterNumber;i++) {

         //this.setState({amountOwed:this.state.amountOwed + 5})
         return <div className="line-item"><span>Additional Design</span><span>$5</span></div>
       }
     }
   }

   submit = (token) => {
     this.props.dispatch(Actions.charge(token, this.state.amountOwed, this.props.user.id, this.state.publishArray))
   }


  render() {
    return (
      <div className="submissions">
        <Header />
        <div className="body-container">
          <div className="copy-container">
            <h2>Finalize Submissions</h2>
            {this.mySubmissions()}
            <hr />
            {this.firstPayment()}
            {this.additionalPayments()}
            <hr />
            <div className="line-item">
              <span>Subtotal</span>
              <span>${this.state.amountOwed}</span>
            </div>
            <StripeCheckout
              token={this.submit}
              stripeKey="pk_test_CjH0FsjusiwdkHtuILI96lTO"
              currency="USD"
              zipCode={true}
            />
          </div>
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  user: state.user,
  mySubmissions: state.mySubmissions
})

export default connect(mapStateToProps)(cssModules(Finalize, style))
