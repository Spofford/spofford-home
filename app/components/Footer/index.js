import React from "react";
import FontAwesome from "react-fontawesome";
import cssModules from 'react-css-modules';
import style from "./style.css";

export const Footer = () => {
  return (
    <footer className='footer-container'>
      <h2>Spofford Design, Inc.</h2>
      <div>
        <h4>Copyright 2018</h4>
      </div>
    </footer>
  )
};

export default cssModules(Footer, style);
