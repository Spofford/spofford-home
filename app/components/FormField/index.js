import React from "react";
import FontAwesome from "react-fontawesome";
import cssModules from 'react-css-modules';
import style from "./style.css";

export class FormField extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      errorMessage: ""
    }

    this.inlineValidation = this.inlineValidation.bind(this);
  }

  inlineValidation(event) {
    var element = document.getElementById(this.props.id)
    var responseObj = {name:event.target.name}
    if (event.target.value == "") {
      element.style = "border-color:red"
      responseObj.bool = false
      this.props.onValidate(responseObj)
    } else {
      if (event.target.name=="email") {
        var re = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
        if (!re.test(event.target.value)) {
          element.style = "border-color:red"
          this.setState({errorMessage: "Not a valid email address"})
          responseObj.bool = false
          this.props.onValidate(responseObj)
        } else {
          element.style = "border-color:#ccc"
          this.setState({errorMessage: ""})
          responseObj.bool = true
          this.props.onValidate(responseObj)
        }
      }
      if (event.target.name=="passwordConfirmation") {
        if (event.target.value != document.getElementById("signup-password").value) {
          element.style = "border-color:red"
          this.setState({errorMessage: "Passwords do not match"})
          responseObj.bool = false
          this.props.onValidate(responseObj)
        } else {
          element.style = "border-color:#ccc"
          this.setState({errorMessage: ""})
          responseObj.bool = true
          this.props.onValidate(responseObj)
        }
      }
      if (event.target.name=="password") {
        if (event.target.value.length < 8) {
          element.style = "border-color:red"
          this.setState({errorMessage: "8 characters required"})
          responseObj.bool = false
          this.props.onValidate(responseObj)
        } else {
          element.style = "border-color:#ccc"
          this.setState({errorMessage: ""})
          responseObj.bool = true
          this.props.onValidate(responseObj)
        }
      }
      if (event.target.name=="firstName") {
        element.style = "border-color:#ccc"
        this.setState({errorMessage: ""})
        responseObj.bool = true
        this.props.onValidate(responseObj)
      }
      if (event.target.name=="lastName") {
        element.style = "border-color:#ccc"
        this.setState({errorMessage: ""})
        responseObj.bool = true
        this.props.onValidate(responseObj)
      }
    }

  }


  render() {
    return (
      <div className="inputGroup">
        <label>{this.props.label}
          <input
            name={this.props.name}
            id={this.props.id}
            className="input"
            type={this.props.type}
            autoComplete={this.props.autocomplete}
            onChange={this.inlineValidation}
          />
          <div className="error-message">{this.state.errorMessage}</div>
        </label>
      </div>
    )
  }
};

export default cssModules(FormField, style);
