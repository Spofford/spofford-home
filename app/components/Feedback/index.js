import React from "react"
import * as contentful from 'contentful'
const ReactMarkdown = require('react-markdown')
import style from "./style.css"
import { BrowserRouter as Router, Route, Link} from 'react-router-dom'
import { default as Concept } from "../Concept"
import HubspotForm from 'react-hubspot-form'

export class Feedback extends React.Component {

  // Added this:
  constructor(props) {
    super(props);

    this.state = {
      model: {},
      children: [],
      concepts: [],
      firstChild: {}
    }

    this.reset = this.reset.bind(this);
  }



  client = contentful.createClient({
    space: 'cahjy08ew1qz',
    accessToken: '37c6ec31a1a6cb3f533f51fa4c4af8fee88e2f910d9879eb79b2d073ae8cc499'
  })

  componentDidMount() {
    this.reset()
    window.addEventListener('hashchange', this.reset);

    const script = document.createElement('script');
    script.src = 'https://js.hsforms.net/forms/v2.js';
    document.body.appendChild(script);

    script.addEventListener('load', () => {
    	if(window.hbspt) {
      	window.hbspt.forms.create({
        	portalId: '4042167',
          formId: 'd1e13228-c396-4338-bbe3-67cd84f53065',
          target: '#hubspotForm'
        })
      }
    });

  }

  componentWillUnmount() {
      window.removeEventListener('hashchange', this.reset);
  }

  reset() {
    window.scrollTo(0,0);
    let query = this.props.match.params.object
    this.fetchModel().then(this.setModel);
    if (query==='start') {
      this.fetchChildren().then(this.setChildren)
    } else if (query==='finish') {

    } else {
      this.fetchConcepts().then(this.setConcepts)
    }
  }

  fetchModel() {
    if (this.props.match.params.object==='start') {
      return this.client.getEntry('3XtovxD6w0AQ6U4ieSiIow')
    } else if (this.props.match.params.object==='finish') {
      return this.client.getEntry('6ofWITems0U0OqGM6W42GU')
    } else {
      return this.client.getEntry(this.props.match.params.object)
    }
  }

  fetchChildren() {
    return this.client.getEntries({ content_type: 'object' })
  }

  fetchConcepts() {
    return this.client.getEntries({
      content_type: 'concept',
      'fields.object.sys.id': this.props.match.params.object
    })
  }

  setModel = response => {
    this.setState({
      model: response.fields
    })
  }

  setChildren = response => {
    this.setState({
      children: response.items,
      firstChild: response.items[0]
    })
  }

  setConcepts = response => {
    this.setState({
      concepts: response.items
    })
  }

  render() {
    if (this.state.model.next) {
      var next = this.state.model.next
    } else {
      var next = 'finish'
    }

    if (this.props.match.params.object==='start') {

      if (this.state.firstChild.sys) {
        var firstChild = this.state.firstChild.sys.id
      }

      return (
      <div className="feedbackBody">
        <h2>{this.state.model.head}</h2>
        <h3>{this.state.model.subhead}</h3>
        <ReactMarkdown source={this.state.model.bodyText} />
        <Link to={'/feedback/le2rOypBxQQ0k68e6oiQQ'}>Start</Link>
      </div>
      )
    } else if (this.props.match.params.object==='finish') {
      return (
        <div className="feedbackBody">
          <h2>{this.state.model.head}</h2>
          <h3>{this.state.model.subhead}</h3>
          <ReactMarkdown source={this.state.model.bodyText} />
          <div id="hubspotForm"></div>
          <Link to='/'>Return to home</Link>
        </div>
      )
    } else {
      return (
        <div className="feedbackBody">
          <h2>{this.state.model.head}</h2>
          <ReactMarkdown source={this.state.model.background} />
          <h2>{this.state.model.problemInsights}</h2>
          <ReactMarkdown source={this.state.model.problem} />
          {this.state.concepts.map(item =>
            <Concept key={item.sys.id} concept={item} />
          )}
          <iframe src={'https://docs.google.com/forms/d/e/' + this.state.model.formId + '/viewform?embedded=true'} width="100%" height="1250">Loading...</iframe>

          <Link to={'/feedback/' + next}>Next</Link>
        </div>
      )
    }
  }
}

export default Feedback
