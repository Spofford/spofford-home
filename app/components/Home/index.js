import React from "react"
import cssModules from "react-css-modules"
import style from "./style.css"
import * as contentful from 'contentful'
import { Link } from 'react-router-dom'
import FontAwesome from "react-fontawesome";
import { connect } from "react-redux"
import { default as Header } from "../Header"
import { default as Footer } from "../Footer"
import Modal from 'react-modal';

const customStyles = {
  content : {
    // left                  : '50%',
    // right                 : 'auto',
    // bottom                : 'auto'
  }
};

Modal.setAppElement('#root')

export class Home extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      model: {},
      posts: [],
      modalIsOpen: false
    }

    this.openModal = this.openModal.bind(this);
    this.closeModal = this.closeModal.bind(this);
  }

  openModal() {
    var self = this
    setTimeout(function(){ self.setState({modalIsOpen: true}) }, 1000);
  }

  closeModal() {
    this.setState({modalIsOpen: false});
  }

  client = contentful.createClient({
    space: 'cahjy08ew1qz',
    accessToken: '37c6ec31a1a6cb3f533f51fa4c4af8fee88e2f910d9879eb79b2d073ae8cc499'
  })

  componentDidMount() {
    /* window.scrollTo(0, 0); */
    this.fetchModel().then(this.setModel);
    this.fetchPosts().then(this.setPosts);

    this.openModal();
  }

  fetchModel = () => this.client.getEntry('2RAz4CbUQwYcIMSgmuQQQ4')

  fetchPosts = () => this.client.getEntries({ content_type: 'blogPost' })

  setModel = response => {
    this.setState({
      model: response.fields
    })
  }

  setPosts = response => {
    this.setState({
      posts: response.items
    })
  }

  render() {
    const myData = [].concat(this.state.posts)
      .sort((a, b) => a.fields.datePublished < b.fields.datePublished)
      .map((item, i) =>
          <div className="postContainer" key={i}>
            <img src={item.fields.primaryImage.fields.file.url} />
            <h2>{item.fields.head}</h2>
            <h3>{item.fields.subhead}</h3>
            <Link to={'/post/' + item.sys.id}>keep reading<FontAwesome name='chevron-right' /></Link>
          </div>
      );

    return (


      <div className="home">
        <div>
          <Modal
            isOpen={this.state.modalIsOpen}
            onRequestClose={this.closeModal}
            style={customStyles}
            closeTimeoutMS={2000}
            contentLabel="Example Modal">
            <div className="modal-container">
              <h2 ref={subtitle => this.subtitle = subtitle}>The Next Generation of New England Furniture</h2>
              <p>I am a modal</p>
              <Link to="/show"><button onClick={this.closeModal} className="green" >LEARN MORE</button></Link>
              <span onClick={this.closeModal}>DISMISS</span>
            </div>
          </Modal>
        </div>
        <div className="hero">
          <img src="https://s3.us-east-2.amazonaws.com/brand-collateral/logo-light-blue-big.svg" />
          <h1>spofford</h1>
        </div>
        <Header headerStart={true} />
        <div className="body-container">
          <div className="copy-container">
            <h2>{this.state.model.head}</h2>
            <h3>{this.state.model.subhead}</h3>
            <Link to="/about">read on<FontAwesome name='chevron-right' /></Link>
          </div>
          <div className="posts">
          <hr />
            <h2>Conversations</h2>
            {myData}
          </div>
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  user: state.user
})

export default connect(mapStateToProps)(cssModules(Home, style))
