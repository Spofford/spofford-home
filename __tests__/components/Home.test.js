import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import configureMockStore from 'redux-mock-store';
import thunk from "redux-thunk";

import Home from '../../app/components/Home';
import { mapStateToProps, mapDispatchToProps } from '../../app/components/Home';

const mockStore = configureMockStore([thunk]);
const store = mockStore();

describe('<Home />', () => {
  let wrapper, store;
  // our mock login function to replace the one provided by mapDispatchToProps
  const mockContentfn = jest.fn();
  const mockPostsfn = jest.fn();
  const mockModalfn = jest.fn();

   beforeEach(() => {
     const initialState = {
       welcomeModal: true
     };
     store = mockStore(initialState);
     // pass the mock function as the login prop
     wrapper = shallow(<Home store={store} getContent={mockContentfn} fetchPosts={mockPostsfn} toggleModal={mockModalfn} />);
   })

   test('renders the component', () => {
     expect(wrapper).toMatchSnapshot();
   });

});
