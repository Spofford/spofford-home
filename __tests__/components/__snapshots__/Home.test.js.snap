// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Home)
    fetchPosts={[MockFunction]}
    getContent={[MockFunction]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    toggleModal={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "content": undefined,
      "fetchPosts": [Function],
      "getContent": [Function],
      "posts": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "toggleModal": [Function],
      "user": undefined,
      "welcomeModal": undefined,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "content": undefined,
        "fetchPosts": [Function],
        "getContent": [Function],
        "posts": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "toggleModal": [Function],
        "user": undefined,
        "welcomeModal": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
